### ISSUE TITLE
Assert Exception vbo.is_withdraw_allowed( now, op.amount )

### ISSUE THREAD

**Issue by abitmore** [cryptonomex/graphene#312]
Unable to get over block 172075. Commit 30ae8e4f3433d4ee500f0b98e30f05e1ebe806ea

Full logs: https://drive.google.com/open?id=0B3xrm70jSHn4MTZKQV9XVDZyQTg

```
2015-09-12T18:51:20 th_a:invoke handle_block         handle_block ] Got block #172075 from network          application.cpp:383
2015-09-12T18:51:20 th_a:invoke handle_block           push_block ] new_block.block_num(): 172075 new_block.id(): 0002a02b392ec29a32bbde7460a81895ab5e4d39          db_block.cpp:97
2015-09-12T18:51:20 th_a:invoke handle_block          _push_block ] Failed to push new block:
10 assert_exception: Assert Exception
vbo.is_withdraw_allowed( now, op.amount ): 
    {}
    th_a  vesting_balance_evaluator.cpp:103 do_evaluate

    {"op":{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}}
    th_a  vesting_balance_evaluator.cpp:109 do_evaluate

    {}
    th_a  evaluator.cpp:42 start_evaluate

    {}
    th_a  db_block.cpp:580 apply_operation

    {"trx":{"ref_block_num":41002,"ref_block_prefix":1338163065,"expiration":"2015-09-12T18:51:45","operations":[[33,{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}]],"extensions":[],"signatures":["1f48e31b530bf16c32b99809c509de9938f639c68bbb523819ecef3aa03d703c2d6cad44e48f56c8aaf091098022a2cd8a860f2e4cab1c16991221482317e14bee"]}}
    th_a  db_block.cpp:563 _apply_transaction

    {"next_block.block_num()":172075}
    th_a  db_block.cpp:468 _apply_block         db_block.cpp:180
2015-09-12T18:51:20 th_a:invoke handle_block         handle_block ] Error when pushing block:
10 assert_exception: Assert Exception
vbo.is_withdraw_allowed( now, op.amount ): 
    {}
    th_a  vesting_balance_evaluator.cpp:103 do_evaluate

    {"op":{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}}
    th_a  vesting_balance_evaluator.cpp:109 do_evaluate

    {}
    th_a  evaluator.cpp:42 start_evaluate

    {}
    th_a  db_block.cpp:580 apply_operation

    {"trx":{"ref_block_num":41002,"ref_block_prefix":1338163065,"expiration":"2015-09-12T18:51:45","operations":[[33,{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}]],"extensions":[],"signatures":["1f48e31b530bf16c32b99809c509de9938f639c68bbb523819ecef3aa03d703c2d6cad44e48f56c8aaf091098022a2cd8a860f2e4cab1c16991221482317e14bee"]}}
    th_a  db_block.cpp:563 _apply_transaction

    {"next_block.block_num()":172075}
    th_a  db_block.cpp:468 _apply_block

    {"new_block":{"previous":"0002a02a79bfc24fc1bcc0db170fb597260c44aa","timestamp":"2015-09-12T18:51:20","witness":"1.6.9","transaction_merkle_root":"eedebcfcebf08658829eaa863a420f0ac36d569e","extensions":[],"witness_signature":"1f767254f096a7223017f32c6e4ca81f0db4f981c37fe43de4f2605a4bf4015ecc2bf90dcb738dee71321d7b3de12dcb179ba7a41c8eb7e5a1333ae10dc1ec0bd9","transactions":[{"ref_block_num":41002,"ref_block_prefix":1338163065,"expiration":"2015-09-12T18:51:45","operations":[[33,{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}]],"extensions":[],"signatures":["1f48e31b530bf16c32b99809c509de9938f639c68bbb523819ecef3aa03d703c2d6cad44e48f56c8aaf091098022a2cd8a860f2e4cab1c16991221482317e14bee"],"operation_results":[[0,{}]]}]}}
    th_a  db_block.cpp:186 _push_block          application.cpp:409
2015-09-12T18:51:20 th_a:invoke handle_block         handle_block ] Got block #172075 from network          application.cpp:383
2015-09-12T18:51:20 th_a:invoke handle_block           push_block ] new_block.block_num(): 172075 new_block.id(): 0002a02b392ec29a32bbde7460a81895ab5e4d39          db_block.cpp:97
2015-09-12T18:51:20 th_a:invoke handle_block          _push_block ] Failed to push new block:
10 assert_exception: Assert Exception
vbo.is_withdraw_allowed( now, op.amount ): 
    {}
    th_a  vesting_balance_evaluator.cpp:103 do_evaluate

    {"op":{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}}
    th_a  vesting_balance_evaluator.cpp:109 do_evaluate

    {}
    th_a  evaluator.cpp:42 start_evaluate

    {}
    th_a  db_block.cpp:580 apply_operation

    {"trx":{"ref_block_num":41002,"ref_block_prefix":1338163065,"expiration":"2015-09-12T18:51:45","operations":[[33,{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}]],"extensions":[],"signatures":["1f48e31b530bf16c32b99809c509de9938f639c68bbb523819ecef3aa03d703c2d6cad44e48f56c8aaf091098022a2cd8a860f2e4cab1c16991221482317e14bee"]}}
    th_a  db_block.cpp:563 _apply_transaction

    {"next_block.block_num()":172075}
    th_a  db_block.cpp:468 _apply_block         db_block.cpp:180
2015-09-12T18:51:20 th_a:invoke handle_block         handle_block ] Error when pushing block:
10 assert_exception: Assert Exception
vbo.is_withdraw_allowed( now, op.amount ): 
    {}
    th_a  vesting_balance_evaluator.cpp:103 do_evaluate

    {"op":{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}}
    th_a  vesting_balance_evaluator.cpp:109 do_evaluate

    {}
    th_a  evaluator.cpp:42 start_evaluate

    {}
    th_a  db_block.cpp:580 apply_operation

    {"trx":{"ref_block_num":41002,"ref_block_prefix":1338163065,"expiration":"2015-09-12T18:51:45","operations":[[33,{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}]],"extensions":[],"signatures":["1f48e31b530bf16c32b99809c509de9938f639c68bbb523819ecef3aa03d703c2d6cad44e48f56c8aaf091098022a2cd8a860f2e4cab1c16991221482317e14bee"]}}
    th_a  db_block.cpp:563 _apply_transaction

    {"next_block.block_num()":172075}
    th_a  db_block.cpp:468 _apply_block

    {"new_block":{"previous":"0002a02a79bfc24fc1bcc0db170fb597260c44aa","timestamp":"2015-09-12T18:51:20","witness":"1.6.9","transaction_merkle_root":"eedebcfcebf08658829eaa863a420f0ac36d569e","extensions":[],"witness_signature":"1f767254f096a7223017f32c6e4ca81f0db4f981c37fe43de4f2605a4bf4015ecc2bf90dcb738dee71321d7b3de12dcb179ba7a41c8eb7e5a1333ae10dc1ec0bd9","transactions":[{"ref_block_num":41002,"ref_block_prefix":1338163065,"expiration":"2015-09-12T18:51:45","operations":[[33,{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.182","owner":"1.2.14540","amount":{"amount":28200000,"asset_id":"1.3.0"}}]],"extensions":[],"signatures":["1f48e31b530bf16c32b99809c509de9938f639c68bbb523819ecef3aa03d703c2d6cad44e48f56c8aaf091098022a2cd8a860f2e4cab1c16991221482317e14bee"],"operation_results":[[0,{}]]}]}}
    th_a  db_block.cpp:186 _push_block          application.cpp:409
...
...
2015-09-12T18:51:25 th_a:invoke handle_block         handle_block ] Got block #172076 from network          application.cpp:383
2015-09-12T18:51:25 th_a:invoke handle_block           push_block ] new_block.block_num(): 172076 new_block.id(): 0002a02cfc60683b0066dd8f8b14e8c08a845235          db_block.cpp:97
2015-09-12T18:51:25 th_a:invoke handle_block           push_block ] Pushing block to fork database that failed to link: 0002a02cfc60683b0066dd8f8b14e8c08a845235, 172076            fork_database.cpp:57
2015-09-12T18:51:25 th_a:invoke handle_block           push_block ] Head: 172075, 0002a02b392ec29a32bbde7460a81895ab5e4d39          fork_database.cpp:58
2015-09-12T18:51:25 th_a:invoke handle_block          _push_block ] Failed to push new block:
10 assert_exception: Assert Exception
_pending_block.previous == next_block.previous: 
    {"pending.prev":"0002a02a79bfc24fc1bcc0db170fb597260c44aa","next.prev":"0002a02b392ec29a32bbde7460a81895ab5e4d39"}
    th_a  db_block.cpp:584 validate_block_header

    {"next_block.block_num()":172076}
    th_a  db_block.cpp:468 _apply_block         db_block.cpp:180
2015-09-12T18:51:25 th_a:invoke handle_block         handle_block ] Error when pushing block:
10 assert_exception: Assert Exception
_pending_block.previous == next_block.previous: 
    {"pending.prev":"0002a02a79bfc24fc1bcc0db170fb597260c44aa","next.prev":"0002a02b392ec29a32bbde7460a81895ab5e4d39"}
    th_a  db_block.cpp:584 validate_block_header

    {"next_block.block_num()":172076}
    th_a  db_block.cpp:468 _apply_block

    {"new_block":{"previous":"0002a02b392ec29a32bbde7460a81895ab5e4d39","timestamp":"2015-09-12T18:51:25","witness":"1.6.85","transaction_merkle_root":"0000000000000000000000000000000000000000","extensions":[],"witness_signature":"2023d61673da2def6bfc59af4bd13bfc9dc08edbe84bb67a12818b652bab2dfb2d0a975873a8dca73a0bb9a22921e29e378f68d7bcc24b2b0918b078046e6f55d9","transactions":[]}}
    th_a  db_block.cpp:186 _push_block          application.cpp:409

```

**Comment by abitmore** [comment]
Happened again on commit f0f12076fa5d80e07758ae5104854efd1dcb748d

```
2015-09-15T10:00:05 th_a:invoke handle_block           push_block ] new_block.block_num(): 200318 new_block.id(): 00030e7e88ec97ea9060
b607ced428c17a6da186                      db_block.cpp:98
2015-09-15T10:00:05 th_a:invoke handle_block          _push_block ] Failed to push new block:
10 assert_exception: Assert Exception
vbo.is_withdraw_allowed( now, op.amount ):
    {}
    th_a  vesting_balance_evaluator.cpp:103 do_evaluate

    {"op":{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.169","owner":"1.2.22302","amount":{"amount":469900000,"asset_id":"1.3.0"}}}
    th_a  vesting_balance_evaluator.cpp:109 do_evaluate

    {}
    th_a  evaluator.cpp:42 start_evaluate

    {}
    th_a  db_block.cpp:581 apply_operation

    {"trx":{"ref_block_num":3708,"ref_block_prefix":1998828999,"expiration":"2015-09-14T23:36:05","operations":[[33,{"fee":{"amount":100000,"asset_id":"1.3.0"},"vesting_balance":"1.13.169","owner":"1.2.22302","amount":{"amount":469900000,"asset_id":"1.3.0"}}]],"extensions":[],"signatures":["1f5c89f834005d003ce773f6cff3d4604343fe2b23fa03e5000940121e4852f3e53c5f2c2b4b01f267819ebd3515315e58104ba9e270e0130cae4eabda4ccf7b16"]}}
    th_a  db_block.cpp:564 _apply_transaction

    {"next_block.block_num()":200318}
    th_a  db_block.cpp:469 _apply_block                 db_block.cpp:181
```

**Comment by theoreticalbts** [comment]
I can't reproduce this, but I'll add more logging in case it re-occurs.

**Comment by theoreticalbts** [comment]
May be related to fork handling problems fixed by e0414d390e684ae59a326dff5496f04813f215c3, re-open if it reoccurs.
